src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs

```diff
@@ -826,6 +826,24 @@ public SyntaxNode Attribute(string name, params SyntaxNode[] attributeArguments)
             return Attribute(name, (IEnumerable<SyntaxNode>)attributeArguments);
         }
 
+        /// <summary>
+        /// Creates an attribute using <see cref="TypeExpression(ITypeSymbol)"/> if the type can be found in the given
+        /// compilation by its metadata name, or <see cref="DottedName(string)"/> if not.
+        /// </summary>
+        internal SyntaxNode Attribute(string name, Compilation compilation, params SyntaxNode[] attributeArguments)
```

Comment:
❔ Why use this instead of the approach used for `PartNotDiscoverableAttribute` in the linked pull request?

Reply:
@CyrusNajmabadi asked me to here: https://github.com/dotnet/roslyn/pull/38090#discussion_r315337705

I don't actually understand it well enough to explain, so... Cyrus?

Reply:
Ah, I see. You should resolve `System.Diagnostics.DebuggerDisplayAttribute` to a type symbol before trying to offer the refactoring. With the type symbol already available, pass that to `generator.Attribute` instead of passing a string to it.

Reply:
It seems like having the existing `string name` overload of this method is a pit of failure then. When would it be the right thing to use?

Reply:
as mentioned in chat, it's the simple 'quick and dirty' entrypoint for people that don't really care about all these in-depth user scenarios.  if i'm just a person writing a simple one-off tool, then these helpers just make me go faster for my task without boggin me down with stuff i don't care about for that use-case.  

For IDE features we do care, so that effectively means there are parts of the API that shouldbe avoided.

Reply:
Ah, it just clicked with me that it's a public API. Thanks for helping me understand!

Reply:
> Ah, I see. You should resolve `System.Diagnostics.DebuggerDisplayAttribute` to a type symbol before trying to offer the refactoring. With the type symbol already available, pass that to `generator.Attribute` instead of passing a string to it. (This is an existing generator method so you won't need to add a new overload.)

I liked the philosophy that even though something somewhere else might be broken (when the type symbol doesn't resolve), we still put best effort toward helping the programmer do what they have in mind rather than bailing on them. And it has the benefit of not requiring an extra semantic model lookup until the refactoring is invoked. But I'm just happy to be getting this in and I don't feel strongly!

Reply:
@jnm2  I like that philosophy too.  However, i like consistency more :)  So, for now, i would just have this check for the existence of the type (like our other type-dependent features do).  This overall philosophy can be revisited in the future if appropriate.